File: src/JobPostingService/DTOs/UpdateJobPostDto.cs
using JobPostingService.Entities;
using System.ComponentModel.DataAnnotations;

namespace JobPostingService.DTOs
{
   public class UpdateJobPostDto
    {
        public string Title { get; set; }
        public string Description { get; set; }
        public int? PaymentAmount { get; set; }
        public DateTime? Deadline { get; set; }
        public string Category { get; set; }
        public LocationDto Location { get; set; }
    }

}

File: backend/src/job-post/dto/update-job-post.dto.ts
import { IsString, IsNumber, IsDate, IsEnum, IsOptional, IsObject } from 'class-validator';
import { Transform } from 'class-transformer';
import { JobPostCategory } from '../../entities/enums/job-post-category.enum';
import { LocationDto } from './create-job-post.dto';

export class UpdateJobPostDto {
  @IsString()
  @IsOptional()
  title?: string;

  @IsString()
  @IsOptional()
  description?: string;

  @IsNumber()
  @IsOptional()
  paymentAmount?: number;

  @IsDate()
  @IsOptional()
  @Transform(({ value }) => value ? new Date(value) : undefined)
  deadline?: Date;

  @IsEnum(JobPostCategory)
  @IsOptional()
  category?: JobPostCategory;

  @IsObject()
  @IsOptional()
  location?: LocationDto;
} 